src/asl.ts

```diff
@@ -2099,6 +2164,137 @@ export class ASL {
     }
   }
 
+  private joinToStateOutput(
```

Comment:
Big function. Could use some documentation

Reply:
I mean at the top. A summary of what syntax it enables. The docs inside are great